File: docs-examples/example-java/src/test/java/io/micronaut/nats/docs/rpc/ProductListener.java
package io.micronaut.nats.docs.rpc;

import io.micronaut.context.annotation.Requires;
// tag::imports[]
import io.micronaut.nats.annotation.NatsListener;
import io.micronaut.nats.annotation.Subject;
// end::imports[]

@Requires(property = "spec.name", value = "RpcUppercaseSpec")
// tag::clazz[]
@NatsListener
public class ProductListener {

    @Subject("product")
    public String toUpperCase(String data) { // <1>
        return data.toUpperCase(); // <2>
    }
}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/nats/docs/rpc/ProductListener.kt
package io.micronaut.nats.docs.rpc

import io.micronaut.context.annotation.Requires
// tag::imports[]
import io.micronaut.nats.annotation.NatsListener
import io.micronaut.nats.annotation.Subject
import java.util.Collections
// end::imports[]

@Requires(property = "spec.name", value = "RpcUppercaseSpec")
// tag::clazz[]
@NatsListener
class ProductListener {

    @Subject("product")
    fun receive(data: String): String { // <1>
        return data.uppercase() // <2>
    }
}
// end::clazz[]
